controller/stats_reporter.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	workQueueDepthStat   = stats.Int64("work_queue_depth", "Depth of the work queue", stats.UnitNone)
+	reconcileCountStat   = stats.Int64("reconcile_count", "Number of reconcile operations", stats.UnitNone)
+	reconcileLatencyStat = stats.Int64("reconcile_latency", "Latency of reconcile operations", stats.UnitMilliseconds)
+
+	reconcilerTagKey tag.Key
+	keyTagKey        tag.Key
+	successTagKey    tag.Key
+)
+
+func init() {
+	var err error
+	// Create the tag keys that will be used to add tags to our measurements.
+	// Tag keys must conform to the restrictions described in
+	// go.opencensus.io/tag/validate.go. Currently those restrictions are:
+	// - length between 1 and 255 inclusive
+	// - characters are printable US-ASCII
+	reconcilerTagKey, err = tag.NewKey("reconciler")
+	if err != nil {
+		panic(err)
+	}
```

Comment:
A fairly common pattern in Go is to define `mustFoo` methods e.g.
```go
func mustNewKey(s string) tag.Key {
   key, err := tag.NewKey(s)
   if err != nil {
      panic(err)
   }
   return key
}
```

One nice thing about this is that you can then write:
```go
var (
	reconcilerTagKey = mustNewTag("reconciler")
	keyTagKey = mustNewKey("key")
	successTagKey = mustNewKey("success")
)
```

Reply:
Awesome, thanks for the suggestion!